internal/hostnetwork/ovs_bridge.go

```diff
@@ -0,0 +1,406 @@
+			if existingPortUUID == portUUID {
+				slog.Debug("port already attached to bridge", "port", interfaceName, "bridge", bridge.Name)
+				return nil
+			}
+		}
+	}
+
+	var operations []ovsdb.Operation
+
+	interfaceNamedUUID := interfaceUUID
+	if interfaceUUID == "" {
+		interfaceOp, err := ovs.Create(
+			&Interface{
+				UUID: "new_interface",
+				Name: interfaceName,
+				Type: "system", // system type for regular interfaces
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to create interface operation: %w", err)
+		}
+		operations = append(operations, interfaceOp...)
+		interfaceNamedUUID = "new_interface"
+	}
+
+	portNamedUUID := portUUID
+	if portUUID == "" {
+		portOp, err := ovs.Create(
+			&Port{
+				UUID:       "new_port",
+				Name:       interfaceName,
+				Interfaces: []string{interfaceNamedUUID},
+			},
+		)
+		if err != nil {
+			return fmt.Errorf("failed to create port operation: %w", err)
+		}
+		operations = append(operations, portOp...)
+		portNamedUUID = "new_port"
+	}
+
+	mutateOp, err := ovs.Where(bridge).Mutate(bridge, model.Mutation{
+		Field:   &bridge.Ports,
+		Mutator: ovsdb.MutateOperationInsert,
+		Value:   []string{portNamedUUID},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create bridge mutate operation: %w", err)
+	}
+	operations = append(operations, mutateOp...)
+
+	slog.Info("executing OVS transaction to attach port to bridge",
+		"interface", interfaceName,
+		"bridge", bridge.Name,
+		"operations_count", len(operations))
+
+	reply, err := ovs.Transact(ctx, operations...)
+	if err != nil {
+		return fmt.Errorf("OVS transaction failed when attaching interface %s to bridge %s: %w", interfaceName, bridge.Name, err)
+	}
+
+	_, err = ovsdb.CheckOperationResults(reply, operations)
+	if err != nil {
+		return fmt.Errorf("OVS operation failed when attaching interface %s to bridge %s (this usually means the interface doesn't exist on the system or OVS can't access it): %w", interfaceName, bridge.Name, err)
+	}
+
+	slog.Info("successfully added interface to bridge", "name", interfaceName, "bridge", bridge.Name)
+
+	return nil
+}
+
+// ensureInternalPortForBridge creates an OVS internal port for the bridge.
+// This creates the Linux network interface that can be used as a master for macvlan/ipvlan.
+func ensureInternalPortForBridge(ctx context.Context, ovs libovsclient.Client, bridgeUUID, bridgeName string) error {
+	// Check if internal port already exists
+	iface := &Interface{Name: bridgeName}
+	err := ovs.Get(ctx, iface)
+	if err == nil {
+		slog.Debug("internal port already exists for bridge", "name", bridgeName, "UUID", iface.UUID)
+		return nil
+	}
+	if !errors.Is(err, libovsclient.ErrNotFound) {
+		return fmt.Errorf("failed to check if internal interface %q exists: %w", bridgeName, err)
+	}
+	slog.Debug("bridge management port does not exist; must create it", "name", bridgeName)
+
+	port := &Port{Name: bridgeName}
+	err = ovs.Get(ctx, port)
```

Comment:
nit: if err := ... ; err != nil

Reply:
ditto; I need to perform more operations with the `err`variable.